### PR TITLE
Fix SSPI ComputeIntegrityCheck with Sign level

### LINKED ISSUE #103461: NegotiateAuthentication.ComputeIntegrityCheck on Windows sets an invalid QOP
### Description

When using `NegotiateAuthentication.ComputeIntegrityCheck` on Windows with a context with `Sign` and not `EncryptAndSign`, it calls [MakeSignature](https://learn.microsoft.com/en-us/windows/win32/api/sspi/nf-sspi-makesignature) with the QoP value of [SECQOP_WRAP_NO_ENCRYPT](https://github.com/dotnet/runtime/blob/5508f791239fd51b1624cc50d7c96e395a7d9fd5/src/libraries/System.Net.Security/src/System/Net/NegotiateAuthenticationPal.Windows.cs#L590). This fails with `SEC_E_INVALID_TOKEN` because as far as I know this QoP setting is for `EncryptMessage` and not `MakeSignature`.

### Reproduction Steps

```csharp
using NegotiateAuthentication client = new(new NegotiateAuthenticationClientOptions()
{
    TargetName = "test/dotnet",
    Package = "Kerberos",
    RequiredProtectionLevel = Enum.Parse<ProtectionLevel>(args[0], true),
    RequireMutualAuthentication = true,
});
using NegotiateAuthentication server = new(new NegotiateAuthenticationServerOptions()
{
    Package = "Kerberos",
});

byte[]? clientOut = client.GetOutgoingBlob(Array.Empty<byte>(), out var clientStatus);
byte[]? serverOut = server.GetOutgoingBlob(clientOut, out var serverStatus);
client.GetOutgoingBlob(serverOut, out clientStatus);

Span<byte> toSign = [0];
ArrayBufferWriter<byte> signatureWriter = new();
client.ComputeIntegrityCheck(toSign, signatureWriter);

Console.WriteLine($"Done: {Convert.ToBase64String(signatureWriter.WrittenSpan)}");
```

In this case I'm running as a domain user that has added `test/dotnet` as a `servicePrincipalName` to its AD account. This allows me to replicate Kerberos authentication in the same process when running as that user.

### Expected behavior

```
$ dotnet run EncryptAndSign

Done: BAQE//////8AAAAAO4pNAKis8sZkw/46ZtDrzQ==
```

Signature is generated

### Actual behavior

```
$ dotnet run Sign

Unhandled exception. System.ComponentModel.Win32Exception (0x80090308): The token supplied to the function is invalid
   at System.Net.NegotiateAuthenticationPal.WindowsNegotiateAuthenticationPal.GetMIC(ReadOnlySpan`1 message, IBufferWriter`1 signature)
   at SigTest.Program.Main(String[] args) in c:\temp\SigTest\Program.cs:line 29
```

Exception, `MakeSignature` fails with `SEC_E_INVALID_TOKEN`.

### Regression?

No

### Known Workarounds

Ensure you set the [NegotiateAuthenticationClientOptions.RequiredProtectionLevel](https://learn.microsoft.com/en-us/dotnet/api/system.net.security.negotiateauthenticationclientoptions.requiredprotectionlevel?view=net-9.0#system-net-security-negotiateauthenticationclientoptions-requiredprotectionlevel) to `EncryptAndSign` and not `Sign`. I don't believe this would cause an issue except that it would just request extra capabilities but I don't believe you should have to request confidentiality if all you need is integrity.

### Configuration

.NET Version - 9.0.100-preview.5.24307.3
OS - Windows Server 2022
Arch - x64

I don't believe this is specific to the OS or arch version. I believe SSPI doesn't accept any QoP values for `MakeSignature`, at least for the builtin authentication providers.

### Other information

_No response_

### PR DESCRIPTION
**Pull request by jborean93**
Fix calling NegotiateAuthentication.ComputeIntegrityCheck on SSPI when the negotiation context was built with ProtectionLevel.Sign. The SECQOP_WRAP_NO_ENCRYPT QoP flag should not be set when calling GetMIC as no encryption is involved and some authentication providers fail when this is set.

Fix #103461

### CODE REVIEW

**Review by filipnavara — APPROVED** [review]

**Comment by jborean93** [comment]
The failure occurred when using Kerberos on SSPI which isn't possible to test on CI. I've verified it manually with the reproducer in #103461 as well as the SshClient I was originally using to test this scenario out.

**Review by rzikm — APPROVED** [review]
LGTM, Thanks!

**Review by wfurt — APPROVED** [review]
LGTM. Is there risk that this can impact typical authentication @filipnavara ?

**Comment by filipnavara** [comment]
> LGTM. Is there risk that this can impact typical authentication @filipnavara ?

TL;DR: No, we don't use this API in the runtime itself.

The API is used in the runtime only for implementing SPNEGO (Unix only managed impl.) and for `NegotiateStream`. In `NegotiateStream` it's only used for NTLM wrapping. This should not affect it at all, and it's covered by unit tests.

**Comment by jborean93** [comment]
Just checking if this is waiting on anything from myself like rebasing the commit from the latest. Just hoping this can be preset in .NET 9 where this API was introduced.

**Comment by rzikm** [comment]
> Just checking if this is waiting on anything from myself like rebasing the commit from the latest. Just hoping this can be preset in .NET 9 where this API was introduced publicly.

Sorry, this has slipped our attention. It looks good to merge. Unfortunately, we are past the .NET 9.0 branch off, so now we need to treat this as regular servicing.

@jborean93 how much does the bug affect you? Is there a workaround?

@karelz this is fixing a bug in new API introduced in .NET 9, how strong business justification do we need to get this serviced?

**Comment by jborean93** [comment]
It’s not the end of the world, I am not familiar with the back porting policies and assumed it would have been easy to do so. If it’s not then that’s fine, especially since it’s not an LTS release.

**Comment by wfurt** [comment]
I think we should try to get it in. I'll try to figure it with @karelz
